libraries/AP_HAL_Linux/AnalogIn_PXF.cpp

```diff
@@ -0,0 +1,142 @@
+/// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+#include <AP_HAL/AP_HAL.h>
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+#include "AnalogIn_PXF.h"
```

Comment:
I'd rather make this AnalogIn_IIO, since it's not PXF only and making it more generic so other boards can benefit from it (as long as they load the drivers).  If you are only aiming PXF, then please also put the PXF/ERLE ifdefs.